modules/case-management/src/test/java/com/ibm/cloud/platform_services/case_management/v1/CaseManagementIT.java

```diff
@@ -0,0 +1,367 @@
+            if (user.getUserId().contentEquals(testUser.userId())) {
+                foundMatchingUser = true;
+                break;
+            }
+        }
+
+        assertTrue(foundMatchingUser);
+    }
+
+    @Test (dependsOnMethods = {"testCreateCase"})
+    public void testFileUpload() {
+
+        String fileName = "test_file.txt";
+
+        List<FileWithMetadata> list = new ArrayList<FileWithMetadata>();
+        byte[] fileBytes = {(byte) 0xde, (byte) 0xad, (byte) 0xbe, (byte) 0xef};
+        InputStream inputStream = new ByteArrayInputStream(fileBytes);
+        FileWithMetadata.Builder builder = new FileWithMetadata.Builder()
+            .filename(fileName)
+            .data(inputStream);
+        FileWithMetadata fileWithMetadata = builder.build();
+        list.add(fileWithMetadata);
+
+        UploadFileOptions uploadFileOptionsModel = new UploadFileOptions.Builder()
+            .caseNumber(newCaseNumber)
+            .file(list)
+            .build();
+
+        Response<Attachment> response = service.uploadFile(uploadFileOptionsModel).execute();
+        Attachment responseObj = response.getResult();
+
+        this.fileAttachmentId = responseObj.getId();
+        assertEquals(fileName, responseObj.getFilename());
+    }
+
+    @Test (dependsOnMethods = {"testCreateCase", "testFileUpload"})
+    public void testDownloadFile() {
+
+        DownloadFileOptions downloadFileOptionsModel = new DownloadFileOptions.Builder()
+            .caseNumber(newCaseNumber)
+            .fileId(fileAttachmentId)
+            .build();
+
+        Response<InputStream> response = service.downloadFile(downloadFileOptionsModel).execute();
+        InputStream responseObj = response.getResult();
+
+        assertNotNull(responseObj);
+    }
+
+    @Test (dependsOnMethods = {"testCreateCase", "testFileUpload", "testDownloadFile"})
+    public void testDeleteFile() {
+
+        DeleteFileOptions deleteFileOptionsModel = new DeleteFileOptions.Builder()
+            .caseNumber(newCaseNumber)
+            .fileId(fileAttachmentId)
+            .build();
+
+        Response<AttachmentList> response = service.deleteFile(deleteFileOptionsModel).execute();
+
+        AttachmentList responseObj = response.getResult();
+
+        // Assert the file attachment list is empty
+        assertEquals(0, responseObj.getAttachments().size());
+    }
+
+    @Test (dependsOnMethods = {"testCreateCase"}, expectedExceptions = {InternalServerErrorException.class})
+    public void testAddResource() {
+
+        // Adding a resource requires a valid CRN (Cloud Resource Name)
+        // CRN's can be retrieved via the Search and Tagging API
+        String invalidCrn = "invalid:crn";
+        AddResourceOptions addResourceOptionsModel = new AddResourceOptions.Builder()
+            .caseNumber(newCaseNumber)
+            .crn(invalidCrn)
+            .note("Test resource")
+            .build();
+
+        service.addResource(addResourceOptionsModel).execute();
```

Comment:
We should assert something here.

Reply:
Nothing to assert here because this test is expected to fail.